platform/newsfeed/src/main.rs

```diff
@@ -0,0 +1,212 @@
+use apache_avro::from_avro_datum;
+use apache_avro::Schema;
+use rdkafka::consumer::{Consumer, StreamConsumer};
+use rdkafka::ClientConfig;
+use rdkafka::Message;
+use regex::Regex;
+use reqwest;
+use scraper::{Html, Selector};
+
+use serde::{Deserialize, Serialize};
+use std::env;
+use tracing_subscriber;
+use tracing_subscriber::filter::EnvFilter;
+use uuid::Uuid;
+// use serde_avro::from_value;
+use tracing::{debug, error, info, instrument, warn};
+use url::Url;
+// use tracing_subscriber::{self, EnvFilter};
+//
+use newsfeed::{ClaudeMessage, ClaudeModel, Prompt, Role};
+
+#[derive(Debug)]
+struct UpstashConfig {
+    bootstrap_server: String,
+    username: String,
+    password: String,
+}
+
+#[instrument]
+async fn create_consumer(upstash: UpstashConfig) -> StreamConsumer {
+    ClientConfig::new()
+        .set("bootstrap.servers", upstash.bootstrap_server)
+        .set("enable.partition.eof", "false")
+        .set("security.protocol", "SASL_SSL")
+        .set("sasl.mechanisms", "SCRAM-SHA-256")
+        .set("sasl.username", &upstash.username)
+        .set("sasl.password", &upstash.password)
+        .set("group.id", format!("chat-{}", Uuid::new_v4()))
+        .create()
+        .expect("Failed to create client")
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+enum RecordNamespace {
+    UpstashAvro,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct RecordDatetime {
+    name: String,
+    #[serde(rename = "type")]
+    field_type: String,
+    doc: String,
+    #[serde(rename = "default")]
+    field_default: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NewsSource {
+    name: String,
+    #[serde(rename = "type")]
+    field_type: String,
+    doc: String,
+    field_default: Option<String>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Fields(
+    #[serde(rename = "record_date_time")] RecordDatetime,
+    #[serde(rename = "news_source")] NewsSource,
+);
+
+#[derive(Debug, Serialize, Deserialize)]
+struct NewsFeedSchema {
+    #[serde(rename = "type")]
+    field_type: String,
+    name: String,
+    namespace: String,
+    fields: Fields,
+}
+
+// #[instrument]
```

Comment:
Still new to Rust but will this commented out line do anything?

Reply:
nope, removing